File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/localization/LocalizationManager.kt
package family.aladdin.android.localization

import android.content.Context
import android.content.res.Configuration
import androidx.compose.runtime.Composable
import androidx.compose.ui.platform.LocalContext
import java.util.Locale

/**
 * 🌍 Localization Manager
 * Управление локализацией приложения
 * Поддержка RU + EN
 */

object LocalizationManager {
    
    // MARK: - Supported Languages
    
    enum class Language(val code: String, val displayName: String, val flag: String) {
        RUSSIAN("ru", "Русский", "🇷🇺"),
        ENGLISH("en", "English", "🇬🇧");
        
        companion object {
            fun fromCode(code: String): Language {
                return values().firstOrNull { it.code == code } ?: RUSSIAN
            }
        }
    }
    
    // MARK: - Current Language
    
    private const val LANGUAGE_KEY = "app_language"
    
    /**
     * Получить текущий язык
     */
    fun getCurrentLanguage(context: Context): Language {
        val prefs = context.getSharedPreferences("aladdin_prefs", Context.MODE_PRIVATE)
        val savedLanguage = prefs.getString(LANGUAGE_KEY, null)
        
        return if (savedLanguage != null) {
            Language.fromCode(savedLanguage)
        } else {
            // Определить язык системы
            val systemLanguage = Locale.getDefault().language
            Language.fromCode(systemLanguage)
        }
    }
    
    /**
     * Сменить язык приложения
     */
    fun changeLanguage(context: Context, language: Language) {
        // Сохранить в SharedPreferences
        val prefs = context.getSharedPreferences("aladdin_prefs", Context.MODE_PRIVATE)
        prefs.edit().putString(LANGUAGE_KEY, language.code).apply()
        
        // Применить язык
        applyLanguage(context, language)
        
        println("✅ Language changed to: ${language.displayName}")
    }
    
    /**
     * Применить язык к Context
     */
    fun applyLanguage(context: Context, language: Language): Context {
        val locale = Locale(language.code)
        Locale.setDefault(locale)
        
        val config = Configuration(context.resources.configuration)
        config.setLocale(locale)
        
        return context.createConfigurationContext(config)
    }
    
    /**
     * Получить локализованную строку
     */
    fun getString(context: Context, stringRes: Int): String {
        return context.getString(stringRes)
    }
    
    /**
     * Получить локализованную строку с параметрами
     */
    fun getString(context: Context, stringRes: Int, vararg formatArgs: Any): String {
        return context.getString(stringRes, *formatArgs)
    }
}

// MARK: - Composable Helper

/**
 * Composable helper для получения строки
 */
@Composable
fun localizedString(stringRes: Int): String {
    val context = LocalContext.current
    return context.getString(stringRes)
}

/**
 * Composable helper для получения строки с параметрами
 */
@Composable
fun localizedString(stringRes: Int, vararg formatArgs: Any): String {
    val context = LocalContext.current
    return context.getString(stringRes, *formatArgs)
}

File: mobile_apps/ALADDIN_iOS/ALADDIN/Core/Localization/LocalizationManager.swift
import Foundation
import SwiftUI

/**
 * 🌍 Localization Manager
 * Управление локализацией приложения
 * Поддержка RU + EN
 */

class LocalizationManager: ObservableObject {
    
    // MARK: - Published Properties
    
    @Published var currentLanguage: Language = .russian
    
    // MARK: - Supported Languages
    
    enum Language: String, CaseIterable {
        case russian = "ru"
        case english = "en"
        
        var displayName: String {
            switch self {
            case .russian: return "Русский"
            case .english: return "English"
            }
        }
        
        var flag: String {
            switch self {
            case .russian: return "🇷🇺"
            case .english: return "🇬🇧"
            }
        }
    }
    
    // MARK: - Init
    
    init() {
        // Определить язык системы
        if let systemLanguage = Locale.current.language.languageCode?.identifier,
           let language = Language(rawValue: systemLanguage) {
            currentLanguage = language
        } else {
            currentLanguage = .russian // По умолчанию русский
        }
    }
    
    // MARK: - Change Language
    
    /**
     * Сменить язык приложения
     */
    func changeLanguage(to language: Language) {
        currentLanguage = language
        UserDefaults.standard.set(language.rawValue, forKey: "appLanguage")
        
        // В production здесь нужно перезапустить UI
        print("✅ Language changed to: \(language.displayName)")
    }
    
    // MARK: - Get Localized String
    
    /**
     * Получить локализованную строку
     */
    func localized(_ key: String) -> String {
        NSLocalizedString(key, comment: "")
    }
    
    /**
     * Получить локализованную строку с параметрами
     */
    func localized(_ key: String, _ arguments: CVarArg...) -> String {
        String(format: NSLocalizedString(key, comment: ""), arguments: arguments)
    }
}

// MARK: - String Extension

extension String {
    /**
     * Удобный helper для локализации
     * Использование: "main.title".localized
     */
    var localized: String {
        NSLocalizedString(self, comment: "")
    }
    
    /**
     * Локализация с параметрами
     * Использование: "family.subtitle".localized(4)
     */
    func localized(_ arguments: CVarArg...) -> String {
        String(format: NSLocalizedString(self, comment: ""), arguments: arguments)
    }
}

// MARK: - LocalizedStringKey Extension

extension LocalizedStringKey {
    /**
     * Создать из строки
     */
    static func key(_ string: String) -> LocalizedStringKey {
        LocalizedStringKey(string)
    }
}




